python/common/python/math/__init__.java

```diff
@@ -0,0 +1,840 @@
+                }
+                break;
+        }
+        java.util.List temp = new java.util.ArrayList<org.python.Object>();
+        temp.add(new org.python.types.Float(mantissa));
+        temp.add(new org.python.types.Int((int) exponent));
+        return new org.python.types.Tuple(temp);
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v", "wObj"}
+    )
+    public static org.python.Object ldexp(org.python.Object v, org.python.Object wObj) {
+        double r = toFloatValue(v);
+        double w = 0.0;
+        if (!(wObj instanceof org.python.types.Int)) {
+            throw new org.python.exceptions.TypeError("Expected an int as second argument to ldexp.");
+        } else {
+            w = toFloatValue(wObj);
+        }
+        if (w < Integer.MIN_VALUE) {
+            w = Integer.MIN_VALUE;
+        } else if (w > Integer.MAX_VALUE) {
+            w = Integer.MAX_VALUE;
+        }
+        double result = Math.scalb(r, (int) w);
+        if (exceptInf(result, r)) {
+            throw new org.python.exceptions.OverflowError("math range error");
+        } else {
+            return new org.python.types.Float(result);
+        }
+
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v", "w"}
+    )
+    public static org.python.Object hypot(org.python.Object v, org.python.Object w) {
+        double x = toFloatValue(v);
+        double y = toFloatValue(w);
+        double mag = Math.hypot(x, y);
+        if (Double.isInfinite(mag) && !(Double.isInfinite(x) || Double.isInfinite(y))) {
+            throw new org.python.exceptions.OverflowError("math range error");
+        }
+        return new org.python.types.Float(mag);
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object radians(org.python.Object v) {
+        return new org.python.types.Float(Math.toRadians(toFloatValue(v)));
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object degrees(org.python.Object v) {
+        return new org.python.types.Float(Math.toDegrees(toFloatValue(v)));
+    }
+
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"arg"}
+    )
+    public static org.python.Object factorial(org.python.Object arg) {
+        double v = 0.0;
+        if (arg instanceof org.python.types.Float) {
+            v = ((org.python.types.Float) arg).value;
+        } else if (arg instanceof org.python.types.Int) {
+            v = (new org.python.types.Float(((org.python.types.Int) arg).value)).value;
+        } else {
+            throw new org.python.exceptions.TypeError("an integer is required (got type " + arg.typeName() + ")");
+        }
+        if ((v - ((int) v)) != 0.0) {
+            throw new org.python.exceptions.ValueError("factorial() only accepts integral values");
+        } else if (v <= 0.0) {
```

Comment:
This seems to be redundant since it overlaps with both `v == 0.0` and `v < 0.0` case, but which behaviour is correct? All these edge cases (including ones like `v == 1.0`) should be tested in the test cases as well.